Meridian59/Data/DataController.cs

```diff
@@ -1902,6 +1913,52 @@ protected void HandleMessage(MessageMessage Message)
             ChatMessages.Add(Message.Message);           
         }
 
+        public void HandleAdminMessage(AdminMessage Message)
+        {
+            if (Message.Message.StartsWith(":< OBJECT"))
+            {
+                HandleAdminShowObjectMessage(Message);
+            }
+        }
+
+        protected void HandleAdminShowObjectMessage(AdminMessage Message)
+        {
+            //Look for object info
+            var regex = new Regex(@":< OBJECT (?<objectnumber>\d*) is CLASS (?<classname>.*)");
+            if (regex.IsMatch(Message.Message))
+            {
+
+
+            }
+            var matches = regex.Matches(Message.Message);
+            if (matches.Count < 1) //We should only get one of these at a time
+            {
+                return;
+            }
+            AdminObject obj = null;
+            foreach (Match match in matches) //but i dont know how to just do if -match will research
+            {
+                obj = new AdminObject(match.Groups["classname"].ToString(), Convert.ToInt32(match.Groups["objectnumber"].ToString()));
+            }
+
+            regex = new Regex(@": (?<property>\w*)\s*=\s(?<datatype>[\w$]*)\s(?<value>\d*)");
+            if (regex.IsMatch((Message.Message)))
+            {
+                BaseList<AdminObjectProperty> props = new BaseList<AdminObjectProperty>();
+                matches = regex.Matches(Message.Message);
+                foreach (Match match in matches)
+                {
+                    props.Add(new AdminObjectProperty(match.Groups["property"].ToString(), match.Groups["datatype"].ToString(), match.Groups["value"].ToString()));
+                }
+                if (obj != null)
+                {
+                    obj.SetProperties(props);
+                    AdminData.WatchObject(obj);
```

Comment:
AdminObject and AdminObjectProperty are currently implementing the INotifyPropertyChanged interface. AdminData is tracking a set of objects at a time. There is an event sent to open the window after the object is "Tracked" (new terminology, replaced watched). This will also be used in the future to poll the server for updates of tracked objects (to refresh the UI). 

There's maybe a better way and I'm eager to hear it.